### PR TITLE
ci: Move Docker labels declaration inside Dockerfile

### PR DESCRIPTION
**Pull request by ggilmore**
Follow up to https://github.com/sourcegraph/sourcegraph/pull/2806

This PR:
1. Moves the `COMMIT_SHA`, etc. labels to be declared inside the Dockerfile via build args instead of in the `docker build ...` command. This is slightly cleaner, and makes it so that people don't need to remember the full metadata key (`org.opencontainer...`) when modifying the build scripts.
1. Adds basic URL, docs metadata labels to the `sourcegraph/alpine` image. All images that use `sourcegraph/alpine` as a base will inherit the value of these labels. cc @slimsag (The sourcegraph/alpine image will need to be re-deployed after this change)

### CODE REVIEW

File: cmd/frontend/Dockerfile [hunk 1]
```diff
@@ -4,6 +4,16 @@
 # ignores.
 
 FROM sourcegraph/alpine:3.9
+
+ARG COMMIT_SHA="unknown"
```
Comment (nicksnyder) [inline cmd/frontend/Dockerfile]: This is repeated a lot. Can/should we put this in our sourcegraph/alpine base image?

Comment (ggilmore) [reply cmd/frontend/Dockerfile]: No, build args are only evaluated at build time. We don't build the sourcegraph/alpine image every time we build a derived image (we just pull down the one on DockerHub).

Comment (ggilmore) [reply cmd/frontend/Dockerfile]: Ultimately, there would be a lot less copy+pasting if we could get rid of these separate `cmd/build.sh` scripts and just generated the appropriate `docker build` command in gen-pipeline.go

Comment (nicksnyder) [reply cmd/frontend/Dockerfile]: Is there anything preventing you from doing that?

Comment (ggilmore) [reply cmd/frontend/Dockerfile]: Take a look at https://github.com/sourcegraph/sourcegraph/blob/master/cmd/server/build.sh for example, our build scripts do a lot of work.

Comment (slimsag) [reply cmd/frontend/Dockerfile]: Our build scripts should ultimately be moved into the `Dockerfile`s themselves. It's not clear to me if you are arguing for something else ("automatically generated docker build commands" sounds complex and not what I would expect)

Comment (slimsag) [reply cmd/frontend/Dockerfile]: urcegraph/sourcegraph/issues/2303

Comment (ggilmore) [reply cmd/frontend/Dockerfile]: I just want the ability to write something like 

```go
args := []string{}
buildArgs := []string{"COMMIT_SHA", "DATE", "VERSION"}
for _, arg := range buildArgs {
	args = append(args, fmt.Sprintf("--build-arg %s", arg)
}

cmds := append(cmds, fmt.Sprintf("docker build %s", strings.Join(args, " "))
```

in gen-pipeline.go

Comment (slimsag) [reply cmd/frontend/Dockerfile]: Got it, that seems reasonable. I think that could be done once #2303 is implemented

**Pull request merged**